receiver/receiverhelper/receiver.go

```diff
@@ -321,27 +323,17 @@ func (mr *metricsReceiver) stopScraping() {
 func (mr *metricsReceiver) closeScrapers(ctx context.Context) error {
 	var errors []error
 
-	for _, scraper := range mr.allBaseScrapers() {
-		if scraper.close == nil {
-			continue
+	for _, scraper := range mr.metricScrapers {
+		if err := scraper.Close(ctx); err != nil {
+			errors = append(errors, err)
 		}
+	}
 
-		if err := scraper.close(ctx); err != nil {
+	for _, scraper := range mr.resourceMetricScrapers {
+		if err := scraper.Close(ctx); err != nil {
 			errors = append(errors, err)
 		}
 	}
 
 	return componenterror.CombineErrors(errors)
 }
-
-func (mr *metricsReceiver) allBaseScrapers() []baseScraper {
```

Comment:
Why remove this?

Reply:
Nobody calls it, and Initialize and Close are on the interface now